### ISSUE TITLE
Snapshot name isn't being honored

### ISSUE THREAD

**Issue by sabeechen** [sabeechen/hassio-google-drive-backup#340]
From @jorojoro66

> I have one issue that I don't understand. When I make a manual snapshot with your tool it perfectly uses the file name as defined in the configuration. But when the automatic daily snapshot is made it doesn't show the name and in Google Drive it just shows as .tar. Under supervisor it shows with filename 7bb7145b. I wonder what will happen with next automatic snapshot. I suppose a filename is generated automatically and random in home assistant, but will it overwrite the snapshot in Drive?
> To go short: why isn't the file name used as defined in the add-on configuration?

**Comment by sabeechen** [comment]
It sound like what's showing up for your snapshot name is the "slug id", which is an internal identifier that Home Assistant gives each snapshot.  It usually shouldn't be the snapshot name.  HA will use the slug ID as the snapshot name if if you don't give it one explicitly when you ask for a snapshot, but the addon should always tell it to use a name.  Do you let the addon create snapshots for you, or do you use an automation somewhere else to create them?

* Make sure the addon, Home Assistant, and the supervisor are up to date.
* When you create a snapshot, do any errors show up in the supervisor logs ("Supervisor" > "System" > "Logs")?
* If none of that seems to work, can you go to "Actions" > "Report an issue" inside the addon and copy the contents of an error report here after you create a snapshot?

**Comment by jorojoro66** [comment]
- Everything up to date
- No errors in log
- No errors, so nothing to copy. Snapshot is being made, just if it's done automatically (every night at 2 am) it doesn't get named as defined in the add-on configuration. If made manual snapshot it gets the right name.

![snapshots](https://user-images.githubusercontent.com/69129758/107103976-61105780-6852-11eb-9e65-6aeb699e4d62.png)

**Comment by sabeechen** [comment]
Can you get me the output of "Actions" > "Report an issue" after an automatic snapshot that has the bad name?

**Comment by jorojoro66** [comment]
Description:
If you have anything else that could help explain what happened, click "Markdown" above and write it here.
Addon version: 0.102.0 Home Assistant Version: 2021.2.1 Supervisor Version: 2021.01.7 Supervisor Channel: stable Hassos Version: 5.10 Docker Version: 19.03.13 Architecture: amd64 Machine: qemux86-64 Date: 2021-02-10T02:10:19.007415+00:00 Timezone: Asia/Bangkok Failure Time: 2021-02-09T19:09:15.144851+00:00 Last Good Sync: 2021-02-10T01:14:59.329723+00:00

Exception:
No error could be identified automatically.
Snapshots:

{
   "HomeAssistant": {
       "snapshots": 4,
       "retained": 1,
       "deletable": 3,
       "name": "HomeAssistant",
       "latest": "2021-02-09T19:00:00Z",
       "size": "2.3 GB"
   },
   "GoogleDrive": {
       "snapshots": 3,
       "retained": 0,
       "deletable": 3,
       "name": "GoogleDrive",
       "latest": "2021-02-09T19:00:00Z",
       "size": "1.7 GB"
   }
}
Config:
{
   "Setting.MAX_SNAPSHOTS_IN_GOOGLE_DRIVE": 3,
   "Setting.DAYS_BETWEEN_SNAPSHOTS": 1.0,
   "Setting.SNAPSHOT_TIME_OF_DAY": "02:00",
   "Setting.SEND_ERROR_REPORTS": true
}
Addon Logs:
02-10 04:14:53 DEBUG [backup.model.syncer] Sync requested by Coordinator
02-10 04:14:53 INFO [backup.model.coordinator] Syncing Snapshots
02-10 04:14:53 DEBUG [backup.ha.harequests] Making Hassio request: http://hassio/snapshots
02-10 04:14:53 DEBUG [backup.drive.driverequests] Requesting refreshed Google Drive credentials
02-10 04:14:54 DEBUG [backup.drive.driverequests] Making Google Drive request: https://www.googleapis.com/drive/v3/files/1dpnDObcOO_0FkBINR5kEh4mO79eEkQ1n/?fields=id%2Cname%2CappProperties%2Csize%2Ctrashed%2CmimeType%2CmodifiedTime%2Ccapabilities%2Cparents&supportsAllDrives=true
02-10 04:14:54 DEBUG [backup.drive.driverequests] Making Google Drive request: https://www.googleapis.com/drive/v3/files/?q=%271dpnDObcOO_0FkBINR5kEh4mO79eEkQ1n%27+in+parents&fields=nextPageToken%2Cfiles%28id%2Cname%2CappProperties%2Csize%2Ctrashed%2CmimeType%2CmodifiedTime%2Ccapabilities%2Cparents%29&pageSize=100&supportsAllDrives=true&includeItemsFromAllDrives=true&corpora=allDrives
02-10 05:14:55 DEBUG [backup.model.syncer] Sync requested by Coordinator
02-10 05:14:55 INFO [backup.model.coordinator] Syncing Snapshots
02-10 05:14:55 DEBUG [backup.ha.harequests] Making Hassio request: http://hassio/snapshots
02-10 05:14:55 DEBUG [backup.drive.driverequests] Requesting refreshed Google Drive credentials
02-10 05:14:55 DEBUG [backup.drive.driverequests] Making Google Drive request: https://www.googleapis.com/drive/v3/files/1dpnDObcOO_0FkBINR5kEh4mO79eEkQ1n/?fields=id%2Cname%2CappProperties%2Csize%2Ctrashed%2CmimeType%2CmodifiedTime%2Ccapabilities%2Cparents&supportsAllDrives=true
02-10 05:14:55 DEBUG [backup.drive.driverequests] Making Google Drive request: https://www.googleapis.com/drive/v3/files/?q=%271dpnDObcOO_0FkBINR5kEh4mO79eEkQ1n%27+in+parents&fields=nextPageToken%2Cfiles%28id%2Cname%2CappProperties%2Csize%2Ctrashed%2CmimeType%2CmodifiedTime%2Ccapabilities%2Cparents%29&pageSize=100&supportsAllDrives=true&includeItemsFromAllDrives=true&corpora=allDrives
02-10 06:14:56 DEBUG [backup.model.syncer] Sync requested by Coordinator
02-10 06:14:56 INFO [backup.model.coordinator] Syncing Snapshots
02-10 06:14:56 DEBUG [backup.ha.harequests] Making Hassio request: http://hassio/snapshots
02-10 06:14:56 DEBUG [backup.drive.driverequests] Requesting refreshed Google Drive credentials
02-10 06:14:56 DEBUG [backup.drive.driverequests] Making Google Drive request: https://www.googleapis.com/drive/v3/files/1dpnDObcOO_0FkBINR5kEh4mO79eEkQ1n/?fields=id%2Cname%2CappProperties%2Csize%2Ctrashed%2CmimeType%2CmodifiedTime%2Ccapabilities%2Cparents&supportsAllDrives=true
02-10 06:14:57 DEBUG [backup.drive.driverequests] Making Google Drive request: https://www.googleapis.com/drive/v3/files/?q=%271dpnDObcOO_0FkBINR5kEh4mO79eEkQ1n%27+in+parents&fields=nextPageToken%2Cfiles%28id%2Cname%2CappProperties%2Csize%2Ctrashed%2CmimeType%2CmodifiedTime%2Ccapabilities%2Cparents%29&pageSize=100&supportsAllDrives=true&includeItemsFromAllDrives=true&corpora=allDrives
02-10 07:14:58 DEBUG [backup.model.syncer] Sync requested by Coordinator
02-10 07:14:58 INFO [backup.model.coordinator] Syncing Snapshots
02-10 07:14:58 DEBUG [backup.ha.harequests] Making Hassio request: http://hassio/snapshots
02-10 07:14:58 DEBUG [backup.drive.driverequests] Requesting refreshed Google Drive credentials
02-10 07:14:58 DEBUG [backup.drive.driverequests] Making Google Drive request: https://www.googleapis.com/drive/v3/files/1dpnDObcOO_0FkBINR5kEh4mO79eEkQ1n/?fields=id%2Cname%2CappProperties%2Csize%2Ctrashed%2CmimeType%2CmodifiedTime%2Ccapabilities%2Cparents&supportsAllDrives=true
02-10 07:14:58 DEBUG [backup.drive.driverequests] Making Google Drive request: https://www.googleapis.com/drive/v3/files/?q=%271dpnDObcOO_0FkBINR5kEh4mO79eEkQ1n%27+in+parents&fields=nextPageToken%2Cfiles%28id%2Cname%2CappProperties%2Csize%2Ctrashed%2CmimeType%2CmodifiedTime%2Ccapabilities%2Cparents%29&pageSize=100&supportsAllDrives=true&includeItemsFromAllDrives=true&corpora=allDrives
02-10 08:14:59 DEBUG [backup.model.syncer] Sync requested by Coordinator
02-10 08:14:59 INFO [backup.model.coordinator] Syncing Snapshots
02-10 08:14:59 DEBUG [backup.ha.harequests] Making Hassio request: http://hassio/snapshots
02-10 08:14:59 DEBUG [backup.drive.driverequests] Requesting refreshed Google Drive credentials
02-10 08:15:00 DEBUG [backup.drive.driverequests] Making Google Drive request: https://www.googleapis.com/drive/v3/files/1dpnDObcOO_0FkBINR5kEh4mO79eEkQ1n/?fields=id%2Cname%2CappProperties%2Csize%2Ctrashed%2CmimeType%2CmodifiedTime%2Ccapabilities%2Cparents&supportsAllDrives=true
02-10 08:15:00 DEBUG [backup.drive.driverequests] Making Google Drive request: https://www.googleapis.com/drive/v3/files/?q=%271dpnDObcOO_0FkBINR5kEh4mO79eEkQ1n%27+in+parents&fields=nextPageToken%2Cfiles%28id%2Cname%2CappProperties%2Csize%2Ctrashed%2CmimeType%2CmodifiedTime%2Ccapabilities%2Cparents%29&pageSize=100&supportsAllDrives=true&includeItemsFromAllDrives=true&corpora=allDrives
Supervisor Logs:
[32m21-02-10 01:30:03 INFO (MainThread) [supervisor.homeassistant.api] Updated Home Assistant API token[0m
[32m21-02-10 01:30:03 INFO (MainThread) [supervisor.homeassistant.api] Updated Home Assistant API token[0m
[31m21-02-10 01:41:29 ERROR (MainThread) [supervisor.homeassistant.api] Error on call http://172.30.32.1:8123/api/config: [0m
[32m21-02-10 01:41:39 INFO (MainThread) [supervisor.api.proxy] Home Assistant WebSocket API error: Received message 257:None is not str[0m
[32m21-02-10 01:41:39 INFO (MainThread) [supervisor.api.proxy] Home Assistant WebSocket API error: Received message 257:None is not str[0m
[32m21-02-10 01:41:39 INFO (MainThread) [supervisor.api.proxy] Home Assistant WebSocket API connection is closed[0m
[32m21-02-10 01:41:39 INFO (MainThread) [supervisor.api.proxy] Home Assistant WebSocket API connection is closed[0m
[31m21-02-10 01:43:05 ERROR (MainThread) [supervisor.homeassistant.api] Error on call http://172.30.32.1:8123/api/config: [0m
[31m21-02-10 01:43:06 ERROR (MainThread) [supervisor.homeassistant.api] Error on call http://172.30.32.1:8123/api/config: [0m
[32m21-02-10 01:45:49 INFO (MainThread) [supervisor.api.proxy] Home Assistant WebSocket API request initialize[0m
[32m21-02-10 01:45:49 INFO (MainThread) [supervisor.api.proxy] Home Assistant WebSocket API request initialize[0m
[32m21-02-10 01:45:49 INFO (MainThread) [supervisor.api.proxy] WebSocket access from a0d7b954_appdaemon[0m
[32m21-02-10 01:45:49 INFO (MainThread) [supervisor.api.proxy] WebSocket access from a0d7b954_nodered[0m
[32m21-02-10 01:45:52 INFO (MainThread) [supervisor.api.proxy] Home Assistant WebSocket API request running[0m
[32m21-02-10 01:45:52 INFO (MainThread) [supervisor.api.proxy] Home Assistant WebSocket API request running[0m
[32m21-02-10 01:49:41 INFO (MainThread) [supervisor.host.info] Updating local host information[0m
[32m21-02-10 01:49:42 INFO (MainThread) [supervisor.host.services] Updating service information[0m
[32m21-02-10 01:49:43 INFO (MainThread) [supervisor.host.network] Updating local network information[0m
[32m21-02-10 01:49:52 INFO (MainThread) [supervisor.host.sound] Updating PulseAudio information[0m
[32m21-02-10 01:49:52 INFO (MainThread) [supervisor.host] Host information reload completed[0m
[32m21-02-10 01:51:06 INFO (MainThread) [supervisor.api.security] /supervisor/logs access from cebe7a76_hassio_google_drive_backup[0m
[32m21-02-10 01:51:07 INFO (MainThread) [supervisor.api.security] /core/logs access from cebe7a76_hassio_google_drive_backup[0m
[32m21-02-10 01:55:38 INFO (MainThread) [supervisor.api.security] /supervisor/logs access from cebe7a76_hassio_google_drive_backup[0m
[32m21-02-10 01:55:38 INFO (MainThread) [supervisor.api.security] /core/logs access from cebe7a76_hassio_google_drive_backup[0m
[32m21-02-10 01:58:37 INFO (MainThread) [supervisor.snapshots] Found 4 snapshot files[0m
[32m21-02-10 01:58:58 INFO (MainThread) [supervisor.api.security] /supervisor/logs access from cebe7a76_hassio_google_drive_backup[0m
[32m21-02-10 01:58:58 INFO (MainThread) [supervisor.api.security] /core/logs access from cebe7a76_hassio_google_drive_backup[0m
[32m21-02-10 02:00:11 INFO (MainThread) [supervisor.homeassistant.api] Updated Home Assistant API token[0m
[32m21-02-10 02:10:19 INFO (MainThread) [supervisor.api.security] /supervisor/logs access from cebe7a76_hassio_google_drive_backup[0m

**Comment by sabeechen** [comment]
Bummer, I don't see any obvious problems there so this must be a bug on my side.  I'm thinking there are three possibilities:

- There is a bug in how I'm reading the snapshot name from Home Assistant
- There is a bug in how I send the snapshot name to Home Assistant when I initially request its creation.
- There is a bug in how I send the snapshot name to Google Drive on upload.

Is the snapshot name only missing in Google Drive, or is it also missing in the addon and in "Supervisor" > "Snapshots"?

Side note: You might want to update the addon and docker on your host system.  Probably not causing this problem but I've had the supervisor crap out on me when I let my docker version get too old.

**Comment by jorojoro66** [comment]
Thanks again. Missing in both! See attached

![addon](https://user-images.githubusercontent.com/69129758/107475161-43baf080-6ba6-11eb-877c-1b0162729023.JPG)
![supervisor](https://user-images.githubusercontent.com/69129758/107475167-461d4a80-6ba6-11eb-962a-20919c14b3ec.JPG)

**Comment by sabeechen** [comment]
Sorry this took me a while to get back on, I had to step away from the project for a bit to manage some personal stuff.

I spent some time digging in to how the addon and the supervisor handle naming, and I couldn't find aything that explains this.  AFACT, the addon always sends a name to the supervisor and if it didn't, the addon would throw an exception that prevents the snapshot from getting created in the first place.  Its extermely peculiar that the snapshot name *does* show up when you create a snapshot manually in the addon, as that uses almost exactly the same code path to construct and send the name.

My best guess is that there is something going on with the supervisor on your machine, otherwise I woudl expect more people to have reported this since you're using the addon with relatively stock settings.  Would it be possible for you to get the supervisor logs right after an automatic snapshot is created?  You could do this by:
* Setting the "Time of day" for snapshots to 1 minute in the future in the addon's settings.
* Waiting a few minutes for the addon to start a snapshot.
* Going to "Supervisor" > "System" > "Logs" and copying the output here (or just use the "Report an issue" thing again in the addon).

But make sure you get the logs within a few minutes of the snapshot starting, otherwise the logs from when the snapshot started might be lost.

**Comment by jorojoro66** [comment]
I had to do one more step. Seems that because my snapshot frequency was 1 time every day, it didn't start a new backup when I changed the time, because the last one was younger than 24 hours. Changing frequency to 0 was no option, because this rules out automatic snapshots. Therefore the only solution was to delete my last snapshot. This immediately started making a new snapshot. And guess what? I got the correct name given..... What I now think is that only my specific time for automatic snapshot, or the full hour might cause the issue of missing the name. For coming night I changed it from 2:00 to 2:07. Let's see what happens.

**Comment by sabeechen** [comment]
Alright, that is super weird.  If this is only happening at certain times of day, I have trouble thinking of what might be behind it.

**Comment by jorojoro66** [comment]
I just checked last night’s backup and it’s named as expected. So the specific time (2:00) or the full hour seems to be the problem. I changed the time to 3:00 now and will report again what happened tomorrow morning.

**Comment by sabeechen** [comment]
Thanks, I appreciate you going through the effort to help me figure this out.

**Comment by jorojoro66** [comment]
Good morning again,

Getting even more interesting now. See screenshot. Yesterday in first instance I decided to check again what would happen if I switched back to 2:00am. Later I decided to change it to 3:00am, to see if it would be caused by the full hour, or the specific 2:00 time. This night I got a 2:00 backup, as well as a 3:00 backup..... The 2:00 backup must have been planned immediately when I first choose this time, right?
Interesting is that the 2:00 backup has no name as defined in the backup schedule, and has a time stamp showing exactly 2:00am. The 3:00am backup does have the correct name, and is timestamped 3:17am. Looks like in the first case time and filename are given at the beginning of the backup and in the second case when finishing the backup? I see the same for the backup that I planned for 2:07 yesterday, it had correct name and timestamp 2:11.
For me the problem is solved now, I will make my backups at 3:00am. But I can imagine that the strange thing that happens at 2:00am keeps you wondering what's the cause.


![Capture](https://user-images.githubusercontent.com/69129758/108933936-93a5b700-767e-11eb-9e45-33dbe515b8ba.JPG)

**Comment by sabeechen** [comment]
It continues to be a peculiar problem :)
I'm tempted to dig into this further, but don't know exactly how I'd attempt to do that.  Maybe we should just say "computers are weird" and  leave it at that?  I'm going to close this issue for now but I'll keep this in mind if I ever see anyone also run into anything similar in the future.  I still suspect there is probably something in the supervisor logs that might explain this, since the addon doesn't do anything differently for different times of day.  Feel free to reopen this bug if there is anything else you think I can help with.

Side note: Your snapshots aren't getting created immediately when you schedule them because of a bug in the version you're running (v0.102.0).  It should start happening exactly on schedule if you update to the latest version of the addon.

**Comment by jorojoro66** [comment]
Thanks. Concerning the filename it's fine for me now, I get the file name when using another time. I understand it keeps spinning in your mind, same for me.
But...
Last night I got two backups again, one at 2:00 without proper file name and one at 3:00 with proper file name. This while my backup schedule is on 3:00 now, so why it still makes a backup at 2:00 now? It's quite disturbing, because it takes double space.

Concerning the version, I have all addons on manual update and always update within a few days (wait if others report issues first). But for this addon the supervisor doesn't report there is a new version available. I just turned it to automatic update and will see if anything happens, but guess not, because the supervisor doesn't detect it. I don't know how I can update manually if no update visible in supervisor, other than probably uninstall and reinstall?
